### PR TITLE
Infra UI cleanup -- and learning resource!

### PR DESCRIPTION
**Pull request by SEAjamieD**
### :nut_and_bolt: Description: What code changed, and why?
This was a rough and quick first pass at correcting code throughout the infrastructure pages where we've noticed a multitude of mistakes begin to present themselves.  While this doesn't cover every piece that needs to be addressed, it shows the depth of the work needed and can hopefully be used as a learning resource as well--both for code authors _and_ for code reviewers!

As you go through this, keep PR #4874 open next to it, so you can see how that revised checklist helps to point you towards things covered here.

### :chains: Related Resources

_msorens' remarks_: Following on Jamie's introduction above, I thought this would be a good place to reiterate guidelines I put together (over a span of years!) on authoring and on reviewing code. 

---
Reviewing code is just as important as writing code. Really. Here are some important reasons why:
- It promotes knowledge transfer.
- It shakes out different ideas/strategies.
- It encourages team consistency and communication.
- It helps you improve your craft.
- It reduces buggy software very cost effectively--as close to inception as possible.

#### As a Code Author

1. Be your own first code reviewer. Go through your code thoroughly and provide pre-review comments to your audience. Also look for things your IDE did to "help" you (reformatting, etc).
2. Distinguish between in-code comments, which describe what is present, and pre-review comments, which describe why it changed.
3. Also use pre-review comments to filter out noise (e.g. "only function xyz changed in this file; the other 95% of the changes were just reformatting").
4. Include only one issue in a code review (i.e. apply Single Responsibility Principle) for focus, traceability, and tidiness. It also makes searching commit histories more straightforward.
5. Exception to above: for those really small or simple–but unrelated–things that are just simpler to include now rather than later, mark them as such by beginning with something like "PR Bonus: ...".
6. Include all of one issue including, but not limited to: user interface changes, application code changes, documentation updates, help text revisions, database schema revisions, unit test additions, configuration file updates, build script changes, installer tweaks... and do not forget the new files you added! (Note that this is not advocating an entire, humongous feature at one go; rather whatever size piece you are working on, make it complete.)
7. Rebase your changes so they are on top of the latest code from the main branch before sending out your review.
8. Verify the four cornerstones: unit tests, lint, build, execution.
9. Make your presentation shine: spell check, no leftover TODO's, no locally tweaked configuration files, no orphaned properties/variables/files, no missing files.
10. Always make changes for a reason: you should be able to justify every character you are changing.

#### As a Code Reviewer

Ultimately your task is to answer the question: do the changes accurately and completely cover the requirements?
1. Understand the requirements of the issue before you dig into the code, including reading the issue and other pertinent background material.
2. If there are UI changes, was a screenshot included?
3. Compare requirements against unit tests (every requirement should be tested)
4. Compare unit tests against the code under test (all code should be tested)
5. Review the unit tests as an ensemble (to look for weaknesses, ambiguities, or gaps)
6. Is each unit test named well? (unambiguous, accurate, complete)
7. Does each unit test evaluate only one thing?
8. Do the unit tests have strong coverage? (equivalence class and boundary value analysis)
9. Is each unit test as simple as possible? (ideally cyclomatic complexity === 1)
10. Does the code under test follow best practices? (SOLID principles, no magic numbers, etc.)
11. Can you get by with less code? (is there a simpler way?)
12. Look beyond the changes, from the low-level (if FooBar is changed to FooBaz, are there instances of Foo-Bar as well?) to the high-level (was the README updated correspondingly?)
13. Is the code self-documenting? (that is the code itself, minimizing comments)

#### References

My Zen of Code Review series:
- Part 1: Pre-Review Comments (https://bit.ly/3imYCoO)
- Part 2: Authoring Best Practices (https://bit.ly/35PJVpi)
- Part 3: The Reviewer's Take (https://bit.ly/2XQQTpS)
- Part 4: Review as if you Own the Code (https://bit.ly/3qsNVnv)
- The Sequel: Better Code Reviews with GIT (https://bit.ly/38SveDW)

---

### :+1: Definition of Done
For this particular PR "Done" means we've effectively begun a cleanup as well as communicated the need for further review.

### :athletic_shoe: How to Build and Test the Change
in hab studio
`build components/automate-ui-devproxy && start_all_services`

in Automate/components/automate-ui folder
`make serve`

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
While many of the fixes are code-based only, much has been showing up in the UI as added white spaces, missing focus rings, 5 to 10 pixel off alignments, etc.
Below is one example

Before
Search button sits outside the Search container with extra white space at the bottom.  Create environment button sits below the place of the search bar.
Screen Shot 2021-03-25 at 5 02 23 PM

After
Screen Shot 2021-03-25 at 5 03 04 PM

After
Accessibility rings for the input and search button, Text visible even when screen very condensed.
SearchBarAfter

### CODE REVIEW

File: components/automate-ui/src/app/entities/cds/cds.effects.ts [hunk 1]
```diff
@@ -114,7 +114,7 @@ export class CdsEffects {
     map((action: DownloadContentItemSuccess) => {
       return new CreateNotification({
         type: Type.info,
-        message: `Content Item "${action.payload.name}" was download`
+        message: `Content Item "${action.payload.name}" was downloaded`
```
Comment (msorens) [inline components/automate-ui/src/app/entities/cds/cds.effects.ts:117]: Author note: grammar

File: components/automate-ui/src/app/entities/clients/client.effects.ts [hunk 2]
```diff
@@ -119,7 +119,7 @@ export class ClientEffects {
       map(({ payload: { name } }: DeleteClientSuccess) => {
         return new CreateNotification({
           type: Type.info,
-          message: `Successfully Deleted Client - ${name}.`
+          message: `Successfully deleted client - ${name}.`
```
Comment (msorens) [inline components/automate-ui/src/app/entities/clients/client.effects.ts:122]: Author note: case consistency -- only the first word should be capitalized.

File: components/automate-ui/src/app/entities/data-bags/data-bag-details.effects.ts [hunk 3]
```diff
@@ -51,7 +51,7 @@ export class DataBagItemsEffects {
       const msg = payload.error.error;
       return new CreateNotification({
         type: Type.error,
-        message: `Could not get infra data bag items: ${msg || payload.error}`
+        message: `Could not get data bag items: ${msg || payload.error}`
```
Comment (msorens) [inline components/automate-ui/src/app/entities/data-bags/data-bag-details.effects.ts:54]: Author note: consistency -- majority of uses omit the "infra" so several occurrences changed.

File: components/automate-ui/src/app/entities/data-bags/data-bags.effects.ts [hunk 4]
```diff
@@ -64,7 +64,7 @@ export class DataBagsEffects {
     map(({ payload: { databag: dataBag } }: CreateDataBagSuccess) => {
       return new CreateNotification({
         type: Type.info,
-        message: `Successfully Created Data Bag ${dataBag.name}.`
+        message: `Successfully created data bag - ${dataBag.name}.`
```
Comment (msorens) [inline components/automate-ui/src/app/entities/data-bags/data-bags.effects.ts:67]: Author note: consistency -- this new pattern is supposed to also include the hyphen.

File: components/automate-ui/src/app/entities/environments/environment.effects.ts [hunk 5]
```diff
@@ -130,7 +130,7 @@ export class EnvironmentEffects {
       filter(({ payload }: CreateEnvironmentFailure) => payload.status !== HttpStatus.CONFLICT),
       map(({ payload }: CreateEnvironmentFailure) => new CreateNotification({
         type: Type.error,
-        message: `Could not create notification: ${payload.error.error || payload}.`
+        message: `Could not create environment: ${payload.error.error || payload}.`
```
Comment (msorens) [inline components/automate-ui/src/app/entities/environments/environment.effects.ts:133]: Author note: copy/paste error

File: components/automate-ui/src/app/entities/jobs/job.effects.ts [hunk 6]
```diff
@@ -58,7 +58,7 @@ export class JobEffects {
     ofType(JobActionTypes.JOB_UPDATE_SUCCESS),
     map(() => new CreateNotification({
       type: Type.info,
-      message: 'Updated a scan job.'
+      message: 'Updated scan job.' // TODO: Add the scan job ID here
```
Comment (msorens) [inline components/automate-ui/src/app/entities/jobs/job.effects.ts:61]: Author note: consistency -- every such message needs to include the ID of the object.

File: components/automate-ui/src/app/modules/infra-proxy/client-details/client-details.component.html [hunk 7]
```diff
@@ -32,8 +32,11 @@
         </chef-tab-selector>
       </chef-page-header>
 
-      <app-reset-client-key [openEvent]="openNotificationModal" 
-      [serverId]="serverId" [orgId]="orgId" [name]="name"></app-reset-client-key>
+      <app-reset-client-key
+        [openEvent]="openNotificationModal"
+        [serverId]="serverId"
+        [orgId]="orgId"
+        [name]="name"></app-reset-client-key>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/client-details/client-details.component.html:39]: Author note: each attribute on a separate line make this easier to comprehend instead of one (hidden by itself), then three on the next line.

File: components/automate-ui/src/app/modules/infra-proxy/create-chef-server-modal/create-chef-server-modal.component.html [hunk 8]
```diff
@@ -10,7 +10,7 @@ <h2 slot="title">Add Chef Server </h2>
           </label>
           <chef-error
           *ngIf="(createForm.get('name').hasError('required') || createForm.get('name').hasError('pattern')) && createForm.get('name').dirty">
-          Display Name is required.
+          Name is required.
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/create-chef-server-modal/create-chef-server-modal.component.html:13]: Author note: consistency -- error message must match actual displayed field name (which occurs a few lines higher in the file).

File: components/automate-ui/src/app/modules/infra-proxy/create-data-bag-modal/create-data-bag-modal.component.html [hunk 9]
```diff
@@ -5,12 +5,12 @@ <h2 slot="title">Create Data Bag</h2>
       <div class="margin">
         <chef-form-field>
           <label>
-            <span class="label">Name</span>
+            <span class="label">Name <span aria-hidden="true">*</span></span>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/create-data-bag-modal/create-data-bag-modal.component.html:8]: Author note: accessibility -- every required field needs an asterisk designated it as required.

File: components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.html [hunk 10]
```diff
@@ -5,12 +5,12 @@ <h2 slot="title">Create Data Bag Item</h2>
...
           <chef-form-field>
             <label>
-              <span class="label">Data Bag Item ID</span>
+              <span class="label">Data Bag Item ID <span aria-hidden="true">*</span></span>
               <input chefInput formControlName="itemId" type="text" id="name-input" (keyup)="handleInput($event)" autocomplete="off"/>
             </label>
             <chef-error
             *ngIf="(createForm.get('itemId').hasError('required') || createForm.get('itemId').hasError('pattern')) && createForm.get('itemId').dirty">
-            Name is required.
+            Data Bag Item ID is required.
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.html:13]: Author note: copy/paste error.

File: components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.ts [hunk 11]
```diff
@@ -123,14 +124,11 @@ export class CreateDatabagItemModalComponent implements OnInit, OnDestroy {
...
   onChangeJSON(event: { target: { value: string } }) {
-    // get value from text area
     const newValue = event.target.value;
     try {
-      // parse it to json
       JSON.parse(newValue);
       this.itemAttrParseError = false;
     } catch (ex) {
-      // set parse error if it fails
       this.itemAttrParseError = true;
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.ts:132]: Author note: Comments should only be present if they add value; these are self-evident, no value added. And having the comments makes the code 50% longer and harder to read and harder to maintain.

File: components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.ts [hunk 12]
```diff
@@ -141,7 +139,7 @@ export class CreateDatabagItemModalComponent implements OnInit, OnDestroy {
       this.attr = JSON.parse(this.createForm.controls['itemAttr'].value.trim());
     }
     this.id = {'id': this.createForm.controls['itemId'].value.trim()};
-    const dataBagItem = {
+    const dataBagItem: DataBagItem = {
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/create-databag-item-modal/create-databag-item-modal.component.ts:142]: Author note: Always, always, always strongly type. More often than not, just adding that will reveal errors in the data instance.

File: components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.html [hunk 13]
```diff
@@ -58,7 +58,7 @@ <h2 slot="title">Create Environment</h2>
           </div>
           <div class="button-bar-box">
             <chef-button tertiary (click)="closeCreateModal()">Cancel</chef-button>
-            <chef-button [disabled]="!constraintArray.length || conflictError"  primary id="create-button-object-modal-1" (click)="createEnvironment()">
+            <chef-button [disabled]="!constraints.length || conflictError"  primary id="create-button-object-modal-1" (click)="createEnvironment()">
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.html:61]: Author note: Make names as long as necessary but no longer. We do not care it is an Array. It could be a Set or a List or a Widget; we do care that it is a bunch of constraints.

File: components/automate-ui/src/app/modules/infra-proxy/create-org-modal/create-org-modal.component.html [hunk 14]
```diff
@@ -60,7 +60,7 @@ <h2 slot="title">Add Chef Organization</h2>
           </label>
           <chef-error
             *ngIf="(createForm.get('admin_user').hasError('required') || createForm.get('admin_user').hasError('pattern')) && createForm.get('admin_user').dirty">
-            Admin user is required.
+            Admin User is required.
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/create-org-modal/create-org-modal.component.html:63]: Author note: Here we do want capitals because it is the name of the input field on screen (which occurs a few lines higher in the file).

File: components/automate-ui/src/app/modules/infra-proxy/data-bags-details/data-bags-details.component.ts [hunk 15]
```diff
@@ -205,7 +205,7 @@ export class DataBagsDetailsComponent implements OnInit, OnDestroy {
 
   public startUpdateDataBagItem(item: DataBagItems, jsonData: Object): void {
     this.dataBagItemName = item.name;
-    this.itemDataJson = JSON.stringify(jsonData, undefined, 4);
+    this.itemDataJson = JSON.stringify(jsonData, null, 4);
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/data-bags-details/data-bags-details.component.ts:208]: Author note: wrong values - the behavior when using undefined is, well, undefined 🤷 ! The correct usage is with null.

File: components/automate-ui/src/app/modules/infra-proxy/edit-environment-attribute-modal/edit-environment-attribute-modal.component.html [hunk 16]
```diff
@@ -10,7 +10,7 @@ <h2 slot="title">Edit Environment {{label}}</h2>
       <hr class="divider-constraint" />
 
       <div *ngIf="label === 'Default'">
-        <form [formGroup]="defaulttAttributeForm">
+        <form [formGroup]="defaultAttributeForm">
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/edit-environment-attribute-modal/edit-environment-attribute-modal.component.html:13]: Author note: typo -- running `make spell` would actually catch that.

File: components/automate-ui/src/app/modules/infra-proxy/edit-environment-attribute-modal/edit-environment-attribute-modal.component.ts [hunk 17]
```diff
@@ -105,9 +105,9 @@ export class EditEnvironmentAttributeModalComponent implements OnChanges, OnInit
       this.cookbookConstraints.forEach((element) => {
         this.selectedCookbookNames.push(element.name);
       });
-      this.selectedCookbookNames.forEach((ele) => {
-        this.constraintKeys.forEach((element, index) => {
-          if (ele === element) {
+      this.selectedCookbookNames.forEach((cookbookName) => {
+        this.constraintKeys.forEach((key, index) => {
+          if (cookbookName === key) {
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/edit-environment-attribute-modal/edit-environment-attribute-modal.component.ts:110]: Author note: code clarity -- much easier to comprehend with different variable names.

File: components/automate-ui/src/app/modules/infra-proxy/edit-environment-attribute-modal/edit-environment-attribute-modal.component.ts [hunk 18]
```diff
@@ -191,57 +191,48 @@ export class EditEnvironmentAttributeModalComponent implements OnChanges, OnInit
     this.updateSuccessful = false;
     this.updateInProgress = true;
 
-    if (this.label === 'Constraints') {
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/edit-environment-attribute-modal/edit-environment-attribute-modal.component.ts:194]: Author note: repeated blocks of code is a code smell -- not only does the reworked block of code eliminate the repeated lines, but by doing so, you can instantly see what is different about the three cases. That is much harder to discern in the original.

File: components/automate-ui/src/app/modules/infra-proxy/edit-environment-attribute-modal/edit-environment-attribute-modal.component.ts [hunk 19]
```diff
@@ -275,8 +266,7 @@ export class EditEnvironmentAttributeModalComponent implements OnChanges, OnInit
   private toDisplay(cookbookVersions: Array<CookbookConstraintGrid> = []) {
     const current = {};
     cookbookVersions.forEach((element) => {
-      current[element.name] =
-        `${element.operator}` + ' ' + `${element.version}`;
+      current[element.name] = `${element.operator} ${element.version}`;
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/edit-environment-attribute-modal/edit-environment-attribute-modal.component.ts:269]: Author note: correcting an inappropriate use of backticks (which interpolate objects into string) -- they should not even have been used at all if string concatenation was being used with it.

File: components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.html [hunk 20]
```diff
@@ -31,8 +31,10 @@
...
           [openEvent]="openEnvironmentModal"
-          [serverId]="serverId" [orgId]="orgId"
-          [name]="name" [environment]="environment"
+          [serverId]="serverId"
+          [orgId]="orgId"
+          [name]="name"
+          [environment]="environment"
           [label]="label"
           [cookbookConstraints]="cookbookConstraints"
           [cookbookVersions]="cookbookVersions">
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.html:40]: Author note: clarity -- do not hide some attributes by putting two on a line just in a couple cases. If short enough, OK to put all on one line. Otherwise, each on a line by itself.

File: components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.html [hunk 21]
```diff
@@ -47,7 +49,7 @@
             <div class="constraint-table">
               <div class="edit">
                 <chef-button class="action" primary (click)="openEditAttributeModal(environment?.default_attributes, 'Constraints')">
-                  <img alt="Edit" src="/assets/img/content and icon.png" />
+                  <img alt="Edit" src="/assets/img/content-and-icon.png" />
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.html:52]: Author note: consistency -- no other file uses spaces in the name. (Besides, having spaces in a filename is asking for maintenance headaches.)

File: components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.html [hunk 22]
```diff
@@ -83,20 +85,21 @@
         </label>
         <div class="attr">
           <div class="expand-collapse">
-            <chef-button class="add-circle" primary (click)="tre.expand()">
+            <chef-button class="add-circle" primary (click)="tree.expand()">
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.html:88]: Author note: typo -- running `make spell` would have caught this.

File: components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.scss [hunk 23]
```diff
@@ -167,7 +167,7 @@ img {
   font-size: 12px;
   line-height: 20px;
   letter-spacing: 0.5px;
-  color: #000000;
+  color: $chef-black;
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/environment-details/environment-details.component.scss:170]: Author note: Do not use color literals in CSS files; always use the provided named constants.

File: components/automate-ui/src/app/modules/infra-proxy/select-box/src/lib/select-box.component.html [hunk 24]
```diff
@@ -6,7 +6,7 @@
                 <div class="input-icons role-list-margin">
                     <mat-icon class="icon">search</mat-icon>
                     <input chefInput *ngIf="!disabled && search" type="text" class="selectbox-search input-field"
-                    [(ngModel)]="leftFilterText" [disabled]="disabled" placeholder="Search.." />
+                    [(ngModel)]="leftFilterText" [disabled]="disabled" placeholder="Search.." /> <!-- TODO: do not use two-way binding! -->
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/select-box/src/lib/select-box.component.html:9]: A key tenet of automate design is One-Way Binding. The `[(ngModel)]` uses two-way binding and needs fixing; just have not done it here.

**Review by msorens — COMMENTED** [review]
Lots of author notes below. You will see a high correlation between these types of fixes and the new suggested PR template in https://github.com/chef/automate/pull/4874

File: components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.html [hunk 25]
```diff
@@ -99,10 +99,10 @@
           <chef-table>
             <chef-thead>
               <chef-tr>
-                <chef-th class="no-border-th">Name</chef-th>
-                <chef-th class="no-border-th">Admin</chef-th>
-                <chef-th class="no-border-th">Projects</chef-th>
-                <chef-th class="three-dot-column no-border-th"></chef-th>
+                <chef-th class="no-border">Name</chef-th>
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.html:102]: Author note: add a component global `no-border` class so the CSS is only written once.

File: components/automate-ui/src/app/modules/infra-proxy/clients/clients.component.scss [hunk 26]
```diff
@@ -1,5 +1,20 @@
...
+
+  app-infra-search-bar {
+    flex-grow: 2;
+  }
+
+  chef-button {
+    margin: 0;
+    margin-left: 16px;
+  }
+}
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/clients/clients.component.scss:16]: Author note:  The Search bar / Create button pairing has shown itself to be used over and over again within the infra page, meaning that instead of duplicating this CSS and pattern, It should be condensed into a single component upon further cleanup.  That will allow us to only maintain one set of css and html.

File: components/automate-ui/src/app/modules/infra-proxy/clients/clients.component.scss [hunk 27]
```diff
@@ -1,5 +1,20 @@
 @import "~styles/variables";
 
+.search-create-container {
+  display: flex;
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/clients/clients.component.scss:4]: Author note: Flexing allows us to easily change the width of our element should the user change the size of their browser window, without disturbing the design and user experience

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.html [hunk 28]
```diff
@@ -1,17 +1,20 @@
...
-    <button chef-button class="search-button primary" primary (click)="handleFiltersClick(search_box.value)">
-      <chef-icon>search</chef-icon>
-    </button>
-  </div>
-  <ng-content></ng-content>
-</div>
+<form role='search'
+  (ngSubmit)="onSubmit(search_box.value)"
+  [class.active]="formActive">
+  <label for="search-filter" aria-hidden="true" class="hidden">
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.html:4]: Author note: For accessibility, all inputs should have a label, even if now viewable on the screen so that screen readers may access them.

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.html [hunk 29]
```diff
@@ -1,17 +1,20 @@
...
-      class="search-input"
-      placeholder="Search {{ placeHolder }}"
-      autocomplete="off">
-    <button chef-button class="search-button primary" primary (click)="handleFiltersClick(search_box.value)">
-      <chef-icon>search</chef-icon>
-    </button>
-  </div>
-  <ng-content></ng-content>
-</div>
+<form role='search'
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.html:1]: Author note: Accessibility best practices tell us that search bars should be in the form tag.

Source: http://web-accessibility.carnegiemuseums.org/code/search/#:~:text=Best%20Practices&text=A%20search%20bar%20should%20be,necessary%20to%20use%20the%20form.&text=Also%2C%20always%20include%20the%20word,and%20in%20the%20submit%20button.

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.scss [hunk 30]
```diff
@@ -1,55 +1,47 @@
 @import "~styles/variables";
 @import "~styles/mixins";
 
-.search-wrapper {
+form {
+  display: flex;
   width: 100%;
+  flex-direction: row;
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.scss:7]: Author note: Thinking of our content in a more flexible manner will allow us to write simpler code that can be viewed properly across multiple screen sizes with fewer CSS.

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.spec.ts [hunk 31]
```diff
@@ -25,4 +25,17 @@ describe('InfraSearchBarComponent', () => {
...
+    // which gives our form an accessibilty ring
+    it('By default, the form should not be active', () => {
+      expect(component.formActive).toBe(false);
+    });
+
+    it('Should active the form when input is focused', () => {
+      component.toggleFocus();
+      expect(component.formActive).toBe(true);
+    });
+  });
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.spec.ts:40]: Author Note: All of our components should have more testing than simply "if" the component exists.  The more tests the better.

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.ts [hunk 32]
```diff
@@ -15,24 +13,17 @@ import {
...
+    this.formActive = !this.formActive;
   }
 
-  handleInput(key, currentText): void {
-    switch (key.toLowerCase()) {
-      case 'enter':
-        this.pressEnter(currentText);
-        break;
-    }
-  }
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.ts:37]: Author note: Unnecessary function when using form element instead for your search bar.

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.ts [hunk 33]
```diff
@@ -15,24 +13,17 @@ import {
...
   @Output() searchButtonClick = new EventEmitter<string>();
-  @ViewChild('search_box', { static: true }) inputField: ElementRef;
 
-  handleFiltersClick(currentText: string): void {
+  onSubmit(currentText: string): void {
     this.searchButtonClick.emit(currentText);
   }
 
-  pressEnter(currentText: string): void {
-    this.searchButtonClick.emit(currentText);
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.ts:28]: Author note: Unnecessary and Repetitive function

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.ts [hunk 34]
```diff
@@ -15,24 +13,17 @@ import {
 
 export class InfraSearchBarComponent {
   inputText = '';
+  formActive = false;
 
   @Input() placeHolder: string;
   @Output() searchButtonClick = new EventEmitter<string>();
-  @ViewChild('search_box', { static: true }) inputField: ElementRef;
 
-  handleFiltersClick(currentText: string): void {
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.ts:23]: Author note: Renamed to better align with what is happening in the form.

**Comment by SEAjamieD** [comment]
Another discovery I came across during this exercise that will need to be examined is the way we are choosing to filter within the infra-search-bar, as we are not catching upper vs lowercase.
Limited Search

File: components/automate-ui/src/app/entities/infra-roles/infra-role.effects.ts [hunk 35]
```diff
@@ -48,7 +48,7 @@ export class InfraRoleEffects {
         const msg = payload.error.error;
         return new CreateNotification({
           type: Type.error,
-          message: `Could not get infra roles: ${msg || payload.error}`
+          message: `Could not get roles: ${msg || payload.error}`
```
Comment (vivek-yadav) [inline components/automate-ui/src/app/entities/infra-roles/infra-role.effects.ts:51]: Would infra roles make more sense, as just roles might be confused with internal roles ?

File: components/automate-ui/src/app/entities/orgs/org.effects.ts [hunk 36]
```diff
@@ -118,7 +118,7 @@ export class OrgEffects {
       map(({ payload: { name } }: DeleteOrgSuccess) => {
         return new CreateNotification({
           type: Type.info,
-          message: `Successfully Deleted Organization ${name}.`
+          message: `Successfully deleted organization ${name}.`
```
Comment (vivek-yadav) [inline components/automate-ui/src/app/entities/orgs/org.effects.ts:121]: is ` - ` needed before ${name}

File: components/automate-ui/src/app/entities/orgs/org.effects.ts [hunk 37]
```diff
@@ -147,7 +147,7 @@ export class OrgEffects {
       ofType(OrgActionTypes.UPDATE_SUCCESS),
       map(({ payload: { org } }: UpdateOrgSuccess) => new CreateNotification({
       type: Type.info,
-      message: `Successfully Updated Organization ${org.name}.`
+      message: `Successfully updated organization ${org.name}.`
```
Comment (vivek-yadav) [inline components/automate-ui/src/app/entities/orgs/org.effects.ts:150]: Is ` - ` needed before `${org.name}` ?

File: components/automate-ui/src/app/entities/policies/policy.effects.ts [hunk 38]
```diff
@@ -97,7 +97,7 @@ export class PolicyEffects {
       const removeStr = removed.length === 1 ? removed[0].displayName : `${removed.length} members`;
       return new CreateNotification({
           type: Type.info,
-          message: `Removed ${removeStr}.`
+          message: `Removed member(s) ${removeStr}.`
```
Comment (vivek-yadav) [inline components/automate-ui/src/app/entities/policies/policy.effects.ts:100]: Is ` - ` needed before `${removeStr}` ?

File: components/automate-ui/src/app/pages/+compliance/+scanner/state/scanner.effects.ts [hunk 39]
```diff
@@ -192,7 +192,7 @@ export class ScannerEffects {
     }),
     map(() => new CreateNotification({
       type: Type.info,
-      message: 'Deleted a scan job.'
+      message: 'Deleted scan job.'
```
Comment (vivek-yadav) [inline components/automate-ui/src/app/pages/+compliance/+scanner/state/scanner.effects.ts:195]: is ID of Deleted scan job needed here ? if yes, we can add a todo for later, here also.

**Review by vivek-yadav — APPROVED** [review]

File: components/automate-ui/src/app/entities/infra-roles/infra-role.effects.ts (hunk 35 above)
Comment (msorens) [reply components/automate-ui/src/app/entities/infra-roles/infra-role.effects.ts:51]: Good question! I considered that, but decided against it for two reasons.
1. Omitting it is more consistent with the rest of the infra pages.
2. More importantly, the context makes it clear which type of role.
That said, if you have a strong preference I am fine to change it because your point is valid.

File: components/automate-ui/src/app/entities/orgs/org.effects.ts (hunk 36 above)
Comment (msorens) [reply components/automate-ui/src/app/entities/orgs/org.effects.ts:121]: Yes! Good eye! 🔍

File: components/automate-ui/src/app/entities/policies/policy.effects.ts (hunk 38 above)
Comment (msorens) [reply components/automate-ui/src/app/entities/policies/policy.effects.ts:100]: OK, the answer here is "no". This is still the old format, the difference is the word "Successfully" is not present here. Jon from UX is well aware of the inconsistencies--I spoke to him about it a couple days ago. Apparently, new code we want to go in with "Successfully <verb> <noun> - <id>" but I purposely did not change this one because it is "old" code.

File: components/automate-ui/src/app/pages/+compliance/+scanner/state/scanner.effects.ts (hunk 39 above)
Comment (msorens) [reply components/automate-ui/src/app/pages/+compliance/+scanner/state/scanner.effects.ts:195]: Yes, should mark with TODO. Every such message should have an ID.

**Review by kalroy — APPROVED** [review]
Thanks for this detailed information.

**Review by vinay033 — APPROVED** [review]
Greate work @SEAjamieD , Thanks for these changes. :)

File: components/automate-ui/src/app/modules/infra-proxy/select-box/src/lib/select-box.component.html (hunk 24 above)
Comment (himanshi-chhabra) [reply components/automate-ui/src/app/modules/infra-proxy/select-box/src/lib/select-box.component.html:9]: Agreed, will fix these changes in #4540.

**Review by himanshi-chhabra — APPROVED** [review]
Changes Look good to me. 
Just one opinion, as for success/update notifications we now have added Successfully text according to new Notification UX, and in some cases, it's not added.
eg: In infra-role: L88 : ```message: `Created role ${name}```
and in databag-item: L71: ```message: `Successfully created data bag item ${id}.````

we can make these changes in this issue: #4873

**Comment by himanshi-chhabra** [comment]
> Another discovery I came across during this exercise that will need to be examined is the way we are choosing to filter within the infra-search-bar, as we are not catching upper vs lowercase.
> Limited Search

@SEAjamieD According to chef-manage, we have a case-sensitive filter within infra-search-bar through **AP**I. I have shared a scenario below files.

G-search-keyword
go-search-keyword
g-search-keyword

**Comment by SEAjamieD** [comment]
> > Another discovery I came across during this exercise that will need to be examined is the way we are choosing to filter within the infra-search-bar, as we are not catching upper vs lowercase.

Thanks @himanshi-chhabra - Is this by design? or should we update this to catch both upper and lowercase?
